src/Service/MailerService.php

```diff
@@ -19,4 +25,45 @@ public function sendConfirmationMail(EmailVerifier $emailVerifier, String $email
                 ->htmlTemplate('registration/confirmation_email.html.twig')
                 ->embed(fopen($logoPath, 'r'), 'img'));
     }
+
+    public function sendInvoice(String $emailSender, String $logoPath, Invoice $invoice, String $filePath) : void
+    {
+        $client = $invoice->getClient();
+        $drivingSchool = $invoice->getDrivingSchool();
+
+        $email = (new TemplatedEmail())
+            ->from(new Address($emailSender, 'BotMailAE'))
+            ->to($client->getEmail())
+            ->subject("Facture")
+            ->htmlTemplate('invoice/mail.html.twig')
+            ->context([
+                'clientName' => $client->getFirstname(),
+                'invoiceName' => $invoice->getName(),
+                'drivingSchoolName' => $drivingSchool->getName()
+                ])
+            ->embed(fopen($logoPath, 'r'), 'img')
+            ->addPart(new DataPart(new File($filePath)))
+        ;
+        $this->mailer->send($email);
+    }
+    public function sendContract(String $emailSender, String $logoPath, Contract $contract, String $filePath) : void
+    {
+        $client = $contract->getClient();
+        $drivingSchool = $contract->getDrivingSchool();
+
+        $email = (new TemplatedEmail())
+            ->from(new Address($emailSender, 'BotMailAE'))
+            ->to($client->getEmail())
+            ->subject("Facture")
+            ->htmlTemplate('contract/mail.html.twig')
+            ->context([
+                'clientName' => $client->getFirstname(),
+                'contratName' => $contract->getName(),
+                'drivingSchoolName' => $drivingSchool->getName()
+            ])
+            ->embed(fopen($logoPath, 'r'), 'img')
+            ->addPart(new DataPart(new File($filePath)))
+        ;
+        $this->mailer->send($email);
+    }
```

Comment:
Both functions are exact the same except the name and the document provided

In this case, create one function that can take contract or invoice as document and type it with `Contract|Invoice $document`

Do not forget to make changes where the functions are called